loggo/loggo.py

```diff
@@ -534,31 +516,28 @@ def log(self, level: int, msg: str, extra: Optional[Dict] = None, safe: bool = F
         if self.stopped:
             return
 
-        # make basic log data from constants and what was passed in
         extra = extra or dict()
-        log_data = {**self.log_data, **extra}
 
         if not safe:
-            log_data = self.sanitise(log_data, use_repr=False)
+            extra = self.sanitise(extra, use_repr=False)
             msg = self.sanitise_msg(msg)
 
-        log_data['log_level'] = str(level)
+        extra.update(dict(level=str(level), loggo=str(True)))
 
         # print or write log lines
-        if self.do_print or self.do_write:
-            trace = extra.get('traceback', '')
-            line = self._build_string(msg, level, trace=trace)
+        trace = extra.get('traceback')
+        if self.do_write or self.do_print:
+            ts = extra.get('timestamp', datetime.now().strftime('%d.%m %Y %H:%M:%S'))
+            line = f'{ts}\t{msg}\t{level}'
+        if trace:
```

Comment:
shouldnt this if-block be inside the previous if-block. It makes no sense to define `line` unless `self.do_write or self.do_print`

Reply:
yeah there was a slight error in logic here, fixed now